### PR TITLE
Use list transformation only when shuffled is True

### PR DESCRIPTION
**Pull request by Tortar**
The list transformations are needed just when shuffling

### CODE REVIEW

File: mesa/time.py [hunk 1]
```diff
@@ -264,8 +267,9 @@ def step(self, shuffle_types: bool = True, shuffle_agents: bool = True) -> None:
             shuffle_agents: If True, the order of execution of each agents in a
                             type group is shuffled.
         """
-        type_keys: list[type[Agent]] = list(self.agents_by_type.keys())
+        type_keys: KeysView[type[Agent]] = self.agents_by_type.keys()
         if shuffle_types:
+            type_keys = list(type_keys)
```
Comment (rht) [inline mesa/time.py:272]: The type doesn't seem to be consistent here, because it was a `KeysView`, then here it becomes a `list`.

Comment (Tortar) [reply mesa/time.py:272]: Made it Iterable

Comment (rht) [reply mesa/time.py:272]: We should enable Mypy in CI so that I don't have to catch this manually.

**Comment by rht** [comment]
LGTM

**Pull request merged**